Looks like all the tests are failing due to my new tests bumping us over the threshold (25)for the `ClassDataAbstractionCoupling` checkstyle:

```
Error: eckstyle] [ERROR] /home/runner/work/kafka/kafka/raft/src/test/java/org/apache/kafka/raft/KafkaNetworkChannelTest.java:68:1: Class Data Abstraction Coupling is 26 (max allowed is 25) classes [AssertionError, BeginQuorumEpochResponse, BeginQuorumEpochResponseData, EndQuorumEpochResponse, EndQuorumEpochResponseData, FetchRequestData.ReplicaState, FetchResponse, FetchResponseData, FetchSnapshotResponse, FetchSnapshotResponseData, KafkaNetworkChannel, MockClient, MockTime, Node, OffsetAndEpoch, RaftRequest.Outbound, StubMetadataUpdater, SupportedVersionRange, Thread, TopicPartition, UpdateRaftVoterResponse, UpdateRaftVoterResponseData, VoteResponse, VoteResponseData, VoteResponseData.PartitionData, VoteResponseData.TopicData]. [ClassDataAbstractionCoupling]
```

Two questions:
* Is it acceptable to just suppress this warning? I assume so, because I see a ton of suppressions for similar test cases across the project
* Is it preferable to suppress in the suppress in the global [checkstyle suppressions file](https://github.com/apache/kafka/blob/trunk/checkstyle/suppressions.xml), or is it preferable to do an inline suppression (I see a [handful](https://github.com/apache/kafka/blob/4fcb8b247ec28b9f6e64f8f663e34f5becf0f25f/tools/src/main/java/org/apache/kafka/tools/reassign/ReassignPartitionsCommand.java#L77) of [these](https://github.com/apache/kafka/blob/4fcb8b247ec28b9f6e64f8f663e34f5becf0f25f/connect/runtime/src/test/java/org/apache/kafka/connect/util/TopicAdminTest.java#L86) [across](https://github.com/apache/kafka/blob/4fcb8b247ec28b9f6e64f8f663e34f5becf0f25f/group-coordinator/src/test/java/org/apache/kafka/coordinator/group/GroupCoordinatorRecordSerdeTest.java#L67) [the](https://github.com/apache/kafka/blob/4fcb8b247ec28b9f6e64f8f663e34f5becf0f25f/storage/src/test/java/org/apache/kafka/tiered/storage/TieredStorageTestBuilder.java#L70) [project](https://github.com/apache/kafka/blob/4fcb8b247ec28b9f6e64f8f663e34f5becf0f25f/clients/src/test/java/org/apache/kafka/clients/consumer/internals/events/ApplicationEventProcessorTest.java#L77))

--- a/checkstyle/suppressions.xml
+++ b/checkstyle/suppressions.xml
@@ -90,7 +90,7 @@
               files="ClientUtils.java"/>
 
     <suppress checks="ClassDataAbstractionCoupling"
-              files="(KafkaConsumer|ConsumerCoordinator|AbstractFetch|KafkaProducer|AbstractRequest|AbstractResponse|TransactionManager|Admin|KafkaAdminClient|MockAdminClient|KafkaRaftClient|KafkaRaftClientTest).java"/>
+              files="(KafkaConsumer|ConsumerCoordinator|AbstractFetch|KafkaProducer|AbstractRequest|AbstractResponse|TransactionManager|Admin|KafkaAdminClient|MockAdminClient|KafkaRaftClient|KafkaRaftClientTest|KafkaNetworkChannelTest).java"/>
     <suppress checks="ClassDataAbstractionCoupling"
               files="(Errors|SaslAuthenticatorTest|AgentTest|CoordinatorTest|NetworkClientTest).java"/>
 

